django/core/cache/backends/filebased.py

```diff
@@ -61,6 +61,14 @@ def set(self, key, value, timeout=DEFAULT_TIMEOUT, version=None):
             if not renamed:
                 os.remove(tmp_path)
 
+    def touch(self, key, timeout=DEFAULT_TIMEOUT, version=None):
+        if not self.has_key(key, version):
```

Comment:
isn't there a race condition here between checking 'has_key' and then get? i.e. the key could be there for `has_key` and disappear before `get` occurs?